### PR TITLE
fix(es/parser): SyntaxError for `export type * from "mod";`

### PR DESCRIPTION
**Pull request by sosukesuzuki**
**Description:**

Following TypeScript programs are invalid ([Playground link](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBUcCGBnOAjTKOAMyggFs4AiAqKgbgChRJYFlUNSLraHGgA)):

```ts
export type * as bar from "mod";
```

```ts
export type * from "mod";
```

But swc TypeScript parser allows the code.

**BREAKING CHANGE:**

N/A

**Related issue (if exists):**

N/A

### CODE REVIEW

File: crates/swc_ecma_parser/tests/typescript/issue-896/input.ts [hunk 1]
```diff
@@ -1,2 +0,0 @@
-// @ts-ignore
-export type * from '../typings'
```
Comment (sosukesuzuki) [inline crates/swc_ecma_parser/tests/typescript/issue-896/input.ts:2]: tsc doesn't throw error for this with `@ts-ignore`. However, tsc doesn't throw error for almost other errors with `@ts-ignore` too. It's not that this error is special.

**Comment by kdy1** [comment]
Seems like tests are failing?

**Comment by sosukesuzuki** [comment]
The action for `Benchmark / Performance regression check` is failed ( https://github.com/swc-project/swc/runs/4569921520?check_suite_focus=true ). But I don't know why..

**Comment by kdy1** [comment]
I meant other ones.
Benchmark will be fixed if you rebase

**Comment by sosukesuzuki** [comment]
Tests of `swc_graph_analyzer` are failing. But, probably it caused by CPU/Memory or networking error.

https://github.com/swc-project/swc/actions/runs/1596414177

> Test (ubuntu-latest, swc_graph_analyzer)
> The hosted runner: GitHub Actions 146 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

File: crates/swc_ecma_parser/src/parser/stmt/module_item.rs [hunk 2]
```diff
@@ -393,6 +393,12 @@ impl<'a, I: Tokens> Parser<I> {
 
         if eat!(self, '*') {
             has_star = true;
+            if self.input.syntax().typescript() && type_only {
+                // export type * from "mod";
+                // or
+                // export type * as foo from "mod";
+                syntax_error!(self, span!(self, start), SyntaxError::TS1383)
```
Comment (kdy1) [inline crates/swc_ecma_parser/src/parser/stmt/module_item.rs:400]: I think we can recover from this error?

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you! 
I think we can recover from the error easily.

File: crates/swc_ecma_parser/src/parser/stmt/module_item.rs (hunk 2 above)
Comment (sosukesuzuki) [reply crates/swc_ecma_parser/src/parser/stmt/module_item.rs:400]: Should we use `self.emit_err`?

Comment (kdy1) [reply crates/swc_ecma_parser/src/parser/stmt/module_item.rs]: Yes

Comment (sosukesuzuki) [reply crates/swc_ecma_parser/src/parser/stmt/module_item.rs]: fixed

**Review by kdy1 — APPROVED** [review]
Thanks!
LGTM.

**Pull request merged**